01_stratified_classification/nbconverted/plot_univariate_fs_results.py

```diff
@@ -0,0 +1,335 @@
+
+# In[1]:
+
+
+import os
+import sys
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+import upsetplot as up
+from venn import generate_petal_labels
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.analysis_utilities as au
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# ### Load results and look at overall performance
+
+# In[2]:
+
+
+results_dir = os.path.join('results', 'univariate_fs', 'pancancer')
+
+
+# In[3]:
+
+
+results_df = au.load_prediction_results_fs(results_dir, cfg.fs_methods)
+
+results_df.loc[
+    (results_df.fs_method == 'mad') & (results_df.n_dims == 100),
+    'fs_method'
+] = 'mad_100'
+results_df.loc[
+    (results_df.fs_method == 'mad') & (results_df.n_dims == 1000),
+    'fs_method'
+] = 'mad_1000'
+
+print(results_df.shape)
+print(results_df.fs_method.unique())
+print(results_df.n_dims.unique())
+results_df.head()
+
+
+# In[4]:
+
+
+compare_df = []
+for fs_method in results_df.fs_method.unique():
+    print(fs_method, file=sys.stderr)
+    compare_df.append(
+        au.compare_control_ind(results_df[results_df.fs_method == fs_method],
+                               metric='aupr', verbose=True)
+          .assign(fs_method=fs_method)
+    )
+compare_df = pd.concat(compare_df)
+
+print(compare_df.shape)
+compare_df.head()
+
+
+# In[5]:
+
+
+sns.set({'figure.figsize': (18, 9)})
+sns.set_context('notebook')
+
+fig, axarr = plt.subplots(2, 3)
+
+for ix, gene in enumerate(compare_df.identifier.unique()):
+    ax = axarr[ix // 3, ix % 3]
+    plot_df = compare_df[compare_df.identifier == gene]
+    sns.boxplot(data=plot_df, x='fs_method', y='delta_aupr', ax=ax)
+    ax.set_title(gene)
+    ax.set_xlabel('Feature selection method')
+    ax.set_ylim(0, 1)
+
+plt.tight_layout()
+
+
+# We can see that the blue box (median f-test) and the purple box (pan-cancer f-test) are reasonably similar in most cases to the green box (no feature selection from 1000 features), suggesting that selecting features by f-statistic doesn't seem to harm performance, at the very least. This is good.
+
+# ### Further analysis of selected features
+# 
+# We want to look at:
+# * Overlap of features in at least one model (for different FS methods)
+# * f-statistic distributions for features in at least one model
+
+# In[6]:
+
+
+# gene to analyze features for
+gene = 'TP53'
+
+id_coefs_info = []
+for identifier, coefs_list in au.generate_nonzero_coefficients_fs(
+        results_dir, cfg.fs_methods):
+    if not identifier.startswith(gene): continue
+    for fold_no, coefs in enumerate(coefs_list):
+        id_coefs_info.append([identifier, fold_no, coefs])
+        
+print(len(id_coefs_info))
+
+
+# In[7]:
+
+
+print(len(id_coefs_info[0]))
+print(id_coefs_info[0])
```

Comment:
This is not interpretable by eye. If that's okay, then no worries, but if it is important, maybe a couple lines of code could format it into a table?

Reply:
I'll try to make this more interpretable - the idea of this cell is just to show what the format of the list is, but I agree it's a bit of a wall of text and colud be clearer.